### PR TITLE
Feature/client broker protocol

### PR DESCRIPTION
**Pull request by apopiak**

### CODE REVIEW

File: source/common/JsonConnection.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+void JsonConnection::setupSignalsAndSlots() {
+    QObject::connect(m_socket, SIGNAL(connected()), this, SLOT(tcpConnected()));
+    QObject::connect(m_socket, SIGNAL(disconnected()), this, SLOT(tcpDisconnected()));
+    QObject::connect(m_socket, SIGNAL(bytesWritten(qint64)), this, SLOT(tcpBytesWritten(qint64)));
+    QObject::connect(m_socket, SIGNAL(readyRead()), this, SLOT(tcpReadyRead()));
+    QObject::connect(m_socket, SIGNAL(error(QAbstractSocket::SocketError)),
+                     this, SLOT(tcpError(QAbstractSocket::SocketError)));
+    QObject::connect(m_socket, SIGNAL(stateChanged(QAbstractSocket::SocketState)),
+                     this, SLOT(tcpStateChanged(QAbstractSocket::SocketState)));
+//    QObject::connect(m_socket, &QAbstractSocket::disconnected,
```
Comment (mrzzzrm) [inline source/common/JsonConnection.cpp]: Disabled this because it made a subsequent connectToHost() impossible (as the socket was deleted)

**Comment by mrzzzrm** [comment]
This includes feature/server-broker-connection

File: source/common/Connection.cpp [hunk 2]
```diff
@@ -50,6 +50,9 @@ void Connection::onJsonData(QJsonDocument doc)
         return;
     }
     auto obj = doc.object();
+
+    qDebug() << "Received: " << obj;
```
Comment (apopiak) [inline source/common/Connection.cpp]: We should think about reducing the amount of logs, I think.

Comment (mrzzzrm) [inline source/common/Connection.cpp]: Removed some as well!

**Pull request merged**